app/models/ext_management_system.rb

```diff
@@ -76,6 +76,11 @@ def self.api_allowed_attributes
   has_many :miq_events,             :as => :target, :dependent => :destroy
   has_many :cloud_subnets, :foreign_key => :ems_id, :dependent => :destroy
 
+  has_many :configuration_scripts,
+           :dependent   => :destroy,
+           :foreign_key => "manager_id",
+           :inverse_of  => :manager
```

Comment:
This is not exactly what I was expecting based on where the conversation was going in https://github.com/ManageIQ/manageiq/pull/17659#discussion_r202030195
but it is probably ok since for things like reporting and expressions it would be available.

I'll prefer if someone from the providers team made the final call on this PR.  cc @blomquisg

Reply:
I talked to @jameswnl about this...  We went round and round about how to handle this best.  In the end, it seems to come back to the way we want to access the data.  We're treating JobTemplates (`ConfigurationScripts`) and Workflows (`ConfigurationWorkflows`) as similar models and we want to be able to retrieve a list containing objects of both types.

However, our current data model has three things subclassing one base class.  Playbooks (`ConfigurationScriptPayload`), JobTemplates, and Workflows all inherit from `ConfigurationScriptBase`.

This PR changes the inheritance chain, so that only Playbooks and JobTemplates inherit from `ConfigurationScriptBase`.  Then, Workflows inherit from JobTemplates (`ConfigurationScript`).  This way, when asking for a list of just `ConfigurationScript`, you can get both JobTemplates and Workflows.

There are other ways to handle this.  We could create a specialized method (`templates_and_workflows`), but that [doesn't have a lot of support](https://github.com/ManageIQ/manageiq/pull/17659#discussion_r201419961).  We could have the callers call two methods `stuff = model.job_templates; stuff << model.workflows`.  We could introduce another model to be the base class of JobTemplates (`ConfigurationScript`) and Workflows (`ConfigurationWorkflows`), say `ConfigurationTemplateBase`.

As it stands now, it's not great.  But, any other direction just seems to add more complication to an already complicated model (and, OMG those names).

If anyone feels strongly about any of the other options for making this work better, I'm open.  But, let's figure it out quickly so we can move forward with this.

Reply:
Can we chat... the OP description is what we agreed on, but then the code doesn't reflect that in that it modifies the base ExtManagementSystem class, and that's where the confusion comes in.  @blomquisg I'm not sure I understand your concerns here based on what you've written.

Reply:
> This PR changes the inheritance chain, so that only Playbooks and JobTemplates inherit from ConfigurationScriptBase. Then, Workflows inherit from JobTemplates (ConfigurationScript). This way, when asking for a list of just ConfigurationScript, you can get both JobTemplates and Workflows.

This is not how the PR is currently implemented. Is it what you suggest to implement by subclassing workflows from job templates? Launching of a job template result in an ansible job. In theory launching of a workflow, a sub type of job template, should also result in an ansible job, but the actual result is NOT.

Reply:
@Fryguy yeah, let's chat ... wanna make sure we're all on the same page.

I agree that the current impl doesn't match OP description...color me still confused.

Reply:
I've update the OP with current class diagram.

Reply:
@Fryguy from class hierarchy perspective, the code is reflecting the OP (OP doesn't describe the relationship side of things 😄 )
Anyway, if we don't want to move the relationship to `EMS` level, we can do like [this wip](https://github.com/ManageIQ/manageiq/pull/17720) which would break the abstraction layers.